forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -0,0 +1,234 @@
+        oauth = OAuthBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the service configurations
+     *
+     * @param block takes the closure to set the values for service configuration.
+     */
+    fun service(block: ServiceBuilder.() -> Unit) {
+        service = ServiceBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the endpoints
+     *
+     * @param block takes the  closure to set the values for urls.
+     */
+    fun urlPath(block: UrlPathBuilder.() -> Unit) {
+        urlPath = UrlPathBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the ssl pinning
+     *
+     * @param block takes the builder closure to set the values for ssl pinning.
+     */
+    fun sslPinning(block: SSLPinningBuilder.() -> Unit) {
+        sslPinning = SSLPinningBuilder().apply(block).build()
+    }
+
+    /**
+     * Build the custom logger
+     *
+     * @param block takes the builder closure to configure custom logger.
+     */
+    fun logger(block: LoggerBuilder.() -> Unit) {
+        logger = LoggerBuilder().apply(block).build()
+    }
+
+    private fun build(): FROptions {
+        return FROptions(server, oauth, service, urlPath, sslPinning, logger)
+    }
+
+}
+
+/**
+ * Data class for the server configurations
+ */
+data class Server(val url: String? = "",
+                  val realm: String? = "root",
+                  val timeout: Int = 30,
+                  val cookieName: String? = "iPlanetDirectoryPro")
+
+/**
+ * Server builder to build the SDK configuration information specific to server
+ */
+class ServerBuilder {
+    var url: String? = ""
+    var realm: String? = "root"
+    var timeout: Int = 30
+    var cookieName: String? = "iPlanetDirectoryPro"
+
+    fun build(): Server = Server(url, realm, timeout, cookieName)
+}
+
+/**
+ * Data class for the OAuth configurations
+ */
+data class OAuth(val oauthClientId: String? = "",
+                 val oauthRedirectUri: String? = "",
+                 val oauthScope: String? = "",
+                 val oauthThresholdSeconds: Long = 30,
+                 val oauthCacheSeconds: Long = 0,
+                 val cookieCacheSeconds: Long = 0)
+
+/**
+ * Oauth builder to build the SDK configuration information specific to oauth
+ */
+class OAuthBuilder {
+    var oauthClientId: String? = ""
+    var oauthRedirectUri: String? = ""
+    var oauthScope: String? = ""
+    var oauthThresholdSeconds: Long = 30
+    var oauthCacheSeconds: Long = 0
+    var cookieCacheSeconds: Long = 0
+
+    fun build() : OAuth = OAuth(oauthClientId, oauthRedirectUri, oauthScope, oauthThresholdSeconds, oauthCacheSeconds, cookieCacheSeconds)
+
+}
+
+/**
+ * Data class for the Service configurations
+ */
+data class Service(val authServiceName: String? = null,
+                   val registrationServiceName: String? = null)
+
+/**
+ * Service builder to build the SDK configuration information specific to services
+ */
+class ServiceBuilder {
+    var authServiceName: String? = null
```

Comment:
In iOS we provide default values authServiceName = "Login" & registrationServiceName = "Registration"

Those are the default trees in AM

Reply:
Android uses this , we can talk about this
    <string name="forgerock_auth_service" translatable="false">UsernamePassword</string>
    <string name="forgerock_registration_service" translatable="false">Registration</string>

Reply:
AM "Login" tree is a default tree in AM always present. I would suggest using Login, as I am not sure that out of the box ID Cloud instances have the UsernamePassword tree

Reply:
Done